src/dynamics/mpas/tests/unit/test_dyn_mpas_procedures.pf

```diff
@@ -1052,4 +1060,336 @@ contains
         test_result = index_unique(-test_data - 10.0_real64)
         @assertEqual(expected_result, test_result)
     end subroutine test_index_unique_by_real_arrays_real64
+
+    @test
+    subroutine test_split_by_empty_string_set()
+        use dyn_mpas_procedures, only: split
+        use funit
+
+        character(*), parameter :: test_string_comma_separated = &
+            ',,Mercury,Venus,,Earth,,,Mars,,Jupiter,Saturn,,Uranus,,,Neptune,,'
+        character(*), parameter :: test_set_comma = ','
+
+        integer :: pos
+
+        ! Test `split` by empty `string`.
+
+        pos = 0
+        call split('', test_set_comma, pos)
+        @assertEqual(1, pos)
+
+        ! Test `split` by empty `string` and optional `back`.
+
+        pos = 1
+        call split('', test_set_comma, pos, back=.true.)
+        @assertEqual(0, pos)
+
+        ! Test `split` by empty `set`.
+
+        pos = 0
+        call split(test_string_comma_separated, '', pos)
+        @assertEqual(66, pos)
+
+        ! Test `split` by empty `set` and optional `back`.
+
+        pos = 66
+        call split(test_string_comma_separated, '', pos, back=.true.)
+        @assertEqual(0, pos)
+    end subroutine test_split_by_empty_string_set
+
+    @test
+    subroutine test_split_by_forward_searches()
+        use dyn_mpas_procedures, only: split
+        use funit
+
+        character(*), parameter :: test_string_comma_separated = &
+            ',,Mercury,Venus,,Earth,,,Mars,,Jupiter,Saturn,,Uranus,,,Neptune,,'
+        character(*), parameter :: test_set_comma = ','
+        character(*), parameter :: test_string_comma_space_separated = &
+            ', Mercury,Venus ,Earth , Mars, Jupiter,Saturn ,Uranus , Neptune, '
```

Comment:
Just to make sure we have covered every edge case, should we include a separator that is just a space in the provided string?  For example:


```suggestion
            ', Mercury,Venus ,Earth , Mars, Jupiter Saturn ,Uranus , Neptune, '
```